packages/makeswift-next-plugin/index.js

```diff
@@ -56,7 +65,115 @@ module.exports = ({ resolveSymlinks } = {}) => (nextConfig = {}) => {
     },
   }
 
-  return withTmInitializer(NEXT_TRANSPILE_MODULES_MODULES, {
-    resolveSymlinks,
-  })(enhancedConfig)
+  await createComponentsManifest()
+
+  return withTmInitializer(NEXT_TRANSPILE_MODULES_MODULES, { resolveSymlinks })(enhancedConfig)
+}
+
+async function createComponentsManifest() {
+  let makeswiftConfig
+  try {
+    // @todo: When will this assumption fail? Monorepos?
+    const filename = path.join(process.cwd(), 'makeswift.config.js')
+    makeswiftConfig = require(filename)
+  } catch (err) {
+    if (err instanceof Error) {
+      if (err.code === 'ENOENT') {
+        makeswiftConfig = defaultMakeswiftConfig
```

Comment:
We should probably _always_ require a Makeswift config, since at the very least they should be providing an API key

Reply:
I think "pit of success" > "supporting many different flavors of configuration".

So, 1 way to do things with very obvious error messages